pages/signup.tsx

```diff
@@ -21,6 +21,22 @@ export default function Signupform(): JSX.Element {
         return url === path;
     };
 
+    async function checkUserSession() {
+        const request = {
+            method: "GET",
+            headers: { "Content-Type": "application/json" },
+        };
+        const response = await fetch("/api/user/me/", request).then(
+            //To get the body do response.json()
+            (response) => response,
+        );
+        if (!response.ok) {
+            router.push("/login").then(() => window.scrollTo(0, 0));
+        }
+    }
+
+    checkUserSession();
+
```

Comment:
Close but not quite.... We want to use getServerProps to handle this on prerender, and it'll be much safer as well (like the class page for example).
```suggestion
```

We want to add the following function to the page:
```typescript
/**
 * getServerSideProps gets the session before this page is rendered
 */
export const getServerSideProps: GetServerSideProps = async (
    context: GetSessionOptions,
) => {
    // obtain the next auth session
    const session = await getSession(context);

    if (!session) {
        return {
            redirect: {
                destination: "/login",
                permanent: false,
            },
        };
    }

    return {
        props: { session },
    };
};
```

We also want to add the session into the wrapper

```typescript
type SignupformProps = {
    session: Record<string, unknown>;
};

/**
 * This is the page that a user will use to either login or register
 * to the SDC platform as a parent of volunteer
 */
export default function Signupform({ session }: SignupformProps): JSX.Element {
    const router = useRouter();
    // hook to hold the url to redirect to
    const [url, setUrl] = useState("");

    const isUrlEmpty = () => {
        return url === "";
    };

    const isUrlPath = (path: string) => {
        return url === path;
    };

    return (
        <Wrapper session={session}>
...
```